fast_llm/engine/training/trainer.py

```diff
@@ -438,6 +453,19 @@ def _prepare_training_state(self) -> None:
             log_main_rank(lambda: f"Loading checkpoint from iteration {last_iteration}...")
             self._load_checkpoint(self._config.training.checkpoint, last_iteration)
 
+        for name, reference_model in self._reference_models.items():
+            pretrained = self._config.reference_models[name].pretrained
+            if pretrained.path is not None and pretrained.model_weights:
+                log_main_rank(f"Loading weights for `{name}` reference model from {pretrained.path}")
+                reference_model.fast_llm_model.load_checkpoint(pretrained)
+            else:
+                log_main_rank(
+                    f"No pretrained checkpoint specified for `{name}` reference model,"
+                    f" using a freshly initialized model...",
+                    log_fn=logger.warning,
+                )
+                reference_model.fast_llm_model.initialize_weights()
```

Comment:
let's just error out here unless we are in debug mode. I don't see a single use case for this other than testing.

Reply:
We don't really have a "debug mode"...